### DISCUSSION TITLE
Extension to edit *.properties files with a standard text editor

### DISCUSSION THREAD

**Discussion by otukatigayim** [microsoft/vscode-discussions#1121] [Extension Development QnA] [answered]
**Content:**

I want to be able to edit *.properties files in the same way as text files.
I would like to implement the following, and if anyone knows an easy way to do it, please let me know.

1. When a *.properties file is clicked in the explorer, convert the Unicode encoded characters written in the file to native encoded characters and display them in the default text editor.
2. Convert the content edited in native encoding in the default text editor to Unicode encoded characters and write it back to the *.properties file.

**Attempted solutions:**

* Using `onDidOpenTextDocument` or `onDidChangeActiveTextEditor` to rewrite the displayed content:
    * This approach marks the file as modified whenever it is opened.
* Adding a conversion process as a CustomTextEditor:
    * I couldn't find a way to display the standard text editor without implementing a WebView.
* Adding a conversion process in a FileSystemProvider:
    * When a *.properties file is clicked in the explorer, the scheme is `file`, so it needs to be reopened with the `properties` scheme in `onDidOpenTextDocument`.
    * However, reopening the file shifts focus to the editor, rendering keyboard operations in the explorer ineffective.

**Alternative approaches:**

If there is a way to meet the above requirements other than the above, please let me know.

**Comment by isidorn** [comment, marked as answer]
That sounds like a fair starting point. Did you build that extension. how did it go?
